draft-ietf-tls-tls13.md

```diff
@@ -3524,12 +3524,11 @@ that:
   types. That is, if a handshake message is split over two or more
   records, there MUST NOT be any other records between them.
 
-- Handshake messages MUST NOT span key changes. Because
-  the ClientHello, EndOfEarlyData, ServerHello, Finished, and KeyUpdate messages
-  can arrive immediately prior to a key change,
-  upon receiving these messages a receiver MUST verify that the end
-  of these messages aligns with a record boundary; if not, then it MUST
-  terminate the connection with an "unexpected_message" alert.
+- Handshake messages MUST NOT span key changes. Implementations MUST verify that
+  all messages immediately preceding a key change align with a record boundary.
+  Because the ClientHello, EndOfEarlyData, ServerHello, Finished, and KeyUpdate
+  messages can immediately precede a key change, implementations MUST send these
+  messages in alignment with a record boundary.
```

Comment:
The title of your commit says "Require implementations to verify record boundary when a key change h...appens rather than on receipt of handshake messages", which I read as saying that you only wanted to enforce the transition on the key change, not at the end of each of these messages (because that it what it says).

I don't care about ClientHello, I care about the key transition that happens after ClientHello.  I don't think that NSS will ever generate a ClientHello that doesn't end with the record, but I don't see any point in adding extra checks.  Especially since they are very hard to add.  If you keep the MUST, there's going to be at least one implementation that is non-compliant (I think, my co-workers sometimes get inappropriately enthusiastic about these things).